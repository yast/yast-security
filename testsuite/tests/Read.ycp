/*
 * YaST2: Modules testsuite
 *
 * Description:
 *   Testsuite for the security module
 *
 * Authors:
 *   Michal Svec <msvec@suse.cz>
 *
 * $Id$
 *
 * testedfiles: Security.ycp PamSettings.ycp Pam.ycp
 */

{

import "Testsuite";
import "Security";

map scr_info = $[
    "sysconfig" : $[
	"security" : $[
	    "PERMISSION_SECURITY"	:	"paranoid",
	    "DISABLE_RESTART_ON_UPDATE" :	"r13",
	    "DISABLE_STOP_ON_REMOVAL"	:	"r14",
	],
	"suseconfig" : $[
	    "CWD_IN_ROOT_PATH"		:	"r2",
	    "CWD_IN_USER_PATH"		:	"r3",
	],
	"displaymanager" : $[
	    "DISPLAYMANAGER_SHUTDOWN"		:	"r3",
	    "DISPLAYMANAGER_REMOTE_ACCESS" :	"r9",
	    "DISPLAYMANAGER_ROOT_LOGIN_REMOTE"	: "r16",
	    "DISPLAYMANAGER_XSERVER_TCP_PORT_6000_OPEN"	: "r17",
	],
	"locate" : $[
	    "RUN_UPDATEDB_AS"		:	"r7",
	],
	"clock" : $[
	    "SYSTOHC"			:	"r12",
	],
	"cron" : $[
	    "SYSLOG_ON_NO_ERROR"	:	"r15",
	],
	"mail" : $[
	    "SMTPD_LISTEN_REMOTE"	:	"r18",
	],
    ],
    "etc" : $[
	"login_defs" : $[
	    "FAIL_DELAY"		:	"l2",
	    "GID_MAX"			:	"l3",
	    "GID_MIN"			:	"l4",
	    "PASS_MAX_DAYS"		:	"l7",
	    "PASS_MIN_DAYS"		:	"l9",
	    "PASS_MIN_LEN"		:	"l10",
	    "PASS_WARN_AGE"		:	"l11",
	    "UID_MAX"			:	"l12",
	    "UID_MIN"			:	"l13",
	    "SYSTEM_UID_MAX"		:	"l14",
	    "SYSTEM_UID_MIN"		:	"l15",
	    "SYSTEM_GID_MAX"		:	"l16",
	    "SYSTEM_GID_MIN"		:	"l17",
	    "USERADD_CMD"		:	"l18",
	    "USERDEL_PRECMD"		:	"l19",
	    "USERDEL_POSTCMD"		:	"l20",
            "ENCRYPT_METHOD"            :       "garbage",
	],
	"inittab" : $[
	    "ca"			:	":ctrlaltdel:/sbin/shutdown -r -t 4 now"
	],
	"sysctl_conf"	: $[
	    "kernel.sysrq"		:	"r8",
	    "net.ipv4.tcp_syncookies"	:	"r9",
	    "net.ipv4.ip_forward"	:	"r10",
	    "net.ipv6.conf.all.forwarding"	:	"r11",
	],
	"polkit-default-privs_local"	: $[
	    "org.freedesktop.upower.hibernate"	: "r12"
	],
    ],
    "target" : $[
        "size" : 1,
        // FileUtils::Exists (ctrl_alt_del_file) returns true, so symlink is called
        "stat"  : $[ 1:2 ]
   ],
    "pam" : $[ "passwd" : $[ "password" : $[
	"pam_unix" : [
	    $[ "arguments" : "nullok use_first_pass use_authtok" ]
	]
    ]]]
];

map E	= $[
    "target" : $[
	"bash_output" : $[]
    ]
];

Testsuite::Test (``(Security::Read()),[scr_info,$[],E],nil);
// read garbage, changed to des
Testsuite::Dump (Security::Settings["PASSWD_ENCRYPTION"]:nil);

scr_info["etc","login_defs","ENCRYPT_METHOD"] = "SHA512";
// reads nil -> not in the Settings
scr_info["etc","login_defs","FAIL_DELAY"] = nil;
Testsuite::Test (``(Security::Read()),[scr_info,$[],E],nil);

Testsuite::Dump (Security::Settings["FAIL_DELAY"]:"none");
Testsuite::Dump (Security::Settings["PASSWD_ENCRYPTION"]:nil);

}
