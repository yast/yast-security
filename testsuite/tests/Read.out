Read	.etc.login_defs.FAIL_DELAY "l2"
Read	.etc.login_defs.GID_MAX "l3"
Read	.etc.login_defs.GID_MIN "l4"
Read	.etc.login_defs.PASS_MAX_DAYS "l7"
Read	.etc.login_defs.PASS_MIN_DAYS "l9"
Read	.etc.login_defs.PASS_WARN_AGE "l11"
Read	.etc.login_defs.UID_MAX "l12"
Read	.etc.login_defs.UID_MIN "l13"
Read	.etc.login_defs.SYS_UID_MAX nil
Read	.etc.login_defs.SYS_UID_MIN nil
Read	.etc.login_defs.SYS_GID_MAX nil
Read	.etc.login_defs.SYS_GID_MIN nil
Read	.etc.login_defs.USERADD_CMD "l18"
Read	.etc.login_defs.USERDEL_PRECMD "l19"
Read	.etc.login_defs.USERDEL_POSTCMD "l20"
Read	.target.size "/etc/sysconfig/clock" 1
Read	.sysconfig.clock.SYSTOHC "r12"
Read	.target.size "/etc/sysconfig/cron" 1
Read	.sysconfig.cron.SYSLOG_ON_NO_ERROR "r15"
Read	.target.size "/etc/sysconfig/displaymanager" 1
Read	.sysconfig.displaymanager.DISPLAYMANAGER_REMOTE_ACCESS "r9"
Read	.target.size "/etc/sysconfig/displaymanager" 1
Read	.sysconfig.displaymanager.DISPLAYMANAGER_ROOT_LOGIN_REMOTE "r16"
Read	.target.size "/etc/sysconfig/displaymanager" 1
Read	.sysconfig.displaymanager.DISPLAYMANAGER_XSERVER_TCP_PORT_6000_OPEN "r17"
Read	.target.size "/etc/sysconfig/displaymanager" 1
Read	.sysconfig.displaymanager.DISPLAYMANAGER_SHUTDOWN "r3"
Read	.target.size "/etc/sysconfig/locate" 1
Read	.sysconfig.locate.RUN_UPDATEDB_AS "r7"
Read	.target.size "/etc/sysconfig/mail" 1
Read	.sysconfig.mail.SMTPD_LISTEN_REMOTE "r18"
Read	.target.size "/etc/sysconfig/security" 1
Read	.sysconfig.security.PERMISSION_SECURITY "paranoid"
Read	.target.size "/etc/sysconfig/services" 1
Read	.sysconfig.services.DISABLE_RESTART_ON_UPDATE nil
Read	.target.size "/etc/sysconfig/services" 1
Read	.sysconfig.services.DISABLE_STOP_ON_REMOVAL nil
Read	.target.size "/etc/sysconfig/suseconfig" 1
Read	.sysconfig.suseconfig.CWD_IN_ROOT_PATH "r2"
Read	.target.size "/etc/sysconfig/suseconfig" 1
Read	.sysconfig.suseconfig.CWD_IN_USER_PATH "r3"
Read	.target.symlink "/etc/systemd/system/ctrl-alt-del.target" nil
Read	.etc.login_defs.ENCRYPT_METHOD "garbage"
Execute	.target.bash_output "/usr/sbin/pam-config -q --cracklib" $[]
Execute	.target.bash_output "/usr/sbin/pam-config -q --pwhistory" $[]
Read	.etc.polkit-default-privs_local."org.freedesktop.upower.hibernate" "r12"
Read	.etc.sysctl_conf."kernel.sysrq" "r8"
Read	.etc.sysctl_conf."net.ipv4.ip_forward" "r10"
Read	.etc.sysctl_conf."net.ipv4.tcp_syncookies" "r9"
Read	.etc.sysctl_conf."net.ipv6.conf.all.forwarding" "r11"
Return	true
Dump	des
Read	.etc.login_defs.FAIL_DELAY nil
Read	.etc.login_defs.GID_MAX "l3"
Read	.etc.login_defs.GID_MIN "l4"
Read	.etc.login_defs.PASS_MAX_DAYS "l7"
Read	.etc.login_defs.PASS_MIN_DAYS "l9"
Read	.etc.login_defs.PASS_WARN_AGE "l11"
Read	.etc.login_defs.UID_MAX "l12"
Read	.etc.login_defs.UID_MIN "l13"
Read	.etc.login_defs.SYS_UID_MAX nil
Read	.etc.login_defs.SYS_UID_MIN nil
Read	.etc.login_defs.SYS_GID_MAX nil
Read	.etc.login_defs.SYS_GID_MIN nil
Read	.etc.login_defs.USERADD_CMD "l18"
Read	.etc.login_defs.USERDEL_PRECMD "l19"
Read	.etc.login_defs.USERDEL_POSTCMD "l20"
Read	.target.size "/etc/sysconfig/clock" 1
Read	.sysconfig.clock.SYSTOHC "r12"
Read	.target.size "/etc/sysconfig/cron" 1
Read	.sysconfig.cron.SYSLOG_ON_NO_ERROR "r15"
Read	.target.size "/etc/sysconfig/displaymanager" 1
Read	.sysconfig.displaymanager.DISPLAYMANAGER_REMOTE_ACCESS "r9"
Read	.target.size "/etc/sysconfig/displaymanager" 1
Read	.sysconfig.displaymanager.DISPLAYMANAGER_ROOT_LOGIN_REMOTE "r16"
Read	.target.size "/etc/sysconfig/displaymanager" 1
Read	.sysconfig.displaymanager.DISPLAYMANAGER_XSERVER_TCP_PORT_6000_OPEN "r17"
Read	.target.size "/etc/sysconfig/displaymanager" 1
Read	.sysconfig.displaymanager.DISPLAYMANAGER_SHUTDOWN "r3"
Read	.target.size "/etc/sysconfig/locate" 1
Read	.sysconfig.locate.RUN_UPDATEDB_AS "r7"
Read	.target.size "/etc/sysconfig/mail" 1
Read	.sysconfig.mail.SMTPD_LISTEN_REMOTE "r18"
Read	.target.size "/etc/sysconfig/security" 1
Read	.sysconfig.security.PERMISSION_SECURITY "paranoid"
Read	.target.size "/etc/sysconfig/services" 1
Read	.sysconfig.services.DISABLE_RESTART_ON_UPDATE nil
Read	.target.size "/etc/sysconfig/services" 1
Read	.sysconfig.services.DISABLE_STOP_ON_REMOVAL nil
Read	.target.size "/etc/sysconfig/suseconfig" 1
Read	.sysconfig.suseconfig.CWD_IN_ROOT_PATH "r2"
Read	.target.size "/etc/sysconfig/suseconfig" 1
Read	.sysconfig.suseconfig.CWD_IN_USER_PATH "r3"
Read	.target.symlink "/etc/systemd/system/ctrl-alt-del.target" nil
Read	.etc.login_defs.ENCRYPT_METHOD "sha512"
Execute	.target.bash_output "/usr/sbin/pam-config -q --cracklib" $[]
Execute	.target.bash_output "/usr/sbin/pam-config -q --pwhistory" $[]
Read	.etc.polkit-default-privs_local."org.freedesktop.upower.hibernate" "r12"
Read	.etc.sysctl_conf."kernel.sysrq" "r8"
Read	.etc.sysctl_conf."net.ipv4.ip_forward" "r10"
Read	.etc.sysctl_conf."net.ipv4.tcp_syncookies" "r9"
Read	.etc.sysctl_conf."net.ipv6.conf.all.forwarding" "r11"
Return	true
Dump	none
